### PR TITLE
#30 Page is no longer need to be reloaded in order to apply CSS

### PR DESCRIPTION
**Pull request by balawama**

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+function runShowMeTheCode(tabId, tab) {
+    if (isGithub(tab)) {
+        chrome.tabs.insertCSS(tabId, {file: 'showMeTheCode.css'});
+    } else {
+        chrome.tabs.insertCSS(tabId, {file: 'disableShowMeTheCode.css'});
+    }
+};
+
+chrome.tabs.onUpdated.addListener(function (tabId, info, tab) {
```
Comment (balawama) [inline src/background.js:31]: It may be possible to attach listener to more appropriate event but I didn't find it

**Comment by balawama** [comment]
zrzut ekranu z 2015-12-09 19-34-54

File: src/background.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    function isBlobView(tab) {
+        return tab.url && tab.url.match(/\/.*blob.*\//);
+    };
+
+    function insertCSS(fileName) {
+        chrome.tabs.insertCSS(tabId, {file: fileName});
+    };
+
+    if (isGitHub(tab)) {
```
Comment (balawama) [inline src/background.js]: we want our code to run only on GitHub

File: src/background.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        if (isBlobView(tab)) {
```
Comment (balawama) [inline src/background.js]: If we open blob view then apply css

File: src/background.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            insertCSS('showMeTheCode.css');
+        } else {
```
Comment (balawama) [inline src/background.js:25]: disable our changes if we're somewhere else

File: src/manifest.json [hunk 5]
```diff
@@ -4,20 +4,18 @@
     "homepage_url": "https://github.com/Viridovix/show-me-the-code-github",
     "version": "1.0",
     "permissions": [
-        "activeTab"
+        "tabs",
+        "notifications",
+        "https://github.com/*"
```
Comment (balawama) [inline src/manifest.json:9]: plugin allowes to mess with everything matching this pattern

File: src/background.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    function isBlobView(tab) {
+        return urlMatches(tab, /.*\/blob\/.*/);
+    };
+
+    function pullRequestFileView(tab) {
+        return urlMatches(tab, /.*\/pull\/.*\/files/);
+    };
+
+    function urlMatches(tab, expression) {
+        return tab.url && tab.url.match(expression);
```
Comment (nowaklukasz) [inline src/background.js:15]: Why do we need `tab.url` check?

Comment (balawama) [inline src/background.js:15]: Chrome executes this code on all pages.
18 gru 2015 6:07 PM "Łukasz Nowak" notifications@github.com napisał(a):

File: src/background.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+function runShowMeTheCode(tabId, tab) {
+    function isGitHub(tab) {
```
Comment (nowaklukasz) [inline src/background.js]: Actually - isn't tab in scope here? I think we can skip passing it to all `isXXX(tab)` functions

File: src/background.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+    function insertCSS(fileName) {
+        chrome.tabs.insertCSS(tabId, {file: fileName});
+    };
+
+    if (isGitHub(tab)) {
+        if (isBlobView(tab) || pullRequestFileView(tab)) {
+            insertCSS('showMeTheCode.css');
+        } else {
+            insertCSS('disableShowMeTheCode.css');
```
Comment (nowaklukasz) [inline src/background.js:26]: Can't we use original style?

File: src/background.js (hunk 7 above)
Comment (balawama) [inline src/background.js]: I'll check
18 gru 2015 6:10 PM "Łukasz Nowak" notifications@github.com napisał(a):

File: src/background.js (hunk 8 above)
Comment (balawama) [inline src/background.js:26]: How can we achieve that? Page is not being reloaded and our CSS stays loaded
18 gru 2015 6:12 PM "Łukasz Nowak" notifications@github.com napisał(a):

File: src/background.js (hunk 6 above)
Comment (nowaklukasz) [inline src/background.js:15]: I get it - but all pages do have url anyway, right? It's the url content matching proper expresion what is interesting for us.
Question is - do we need `tab.url` or `tab.url.match(expression)` is sufficient?

Comment (balawama) [inline src/background.js:15]: This check is needed for chrome configuration page
18 gru 2015 6:14 PM "Łukasz Nowak" notifications@github.com napisał(a):

File: src/background.js (hunk 8 above)
Comment (nowaklukasz) [inline src/background.js:26]: No idea, so :+1:

File: src/background.js (hunk 6 above)
Comment (nowaklukasz) [inline src/background.js:15]: Ack

**Comment by nowaklukasz** [comment]
:+1:

File: src/background.js (hunk 7 above)
Comment (balawama) [inline src/background.js]: `tab` removed

**Pull request merged**